tidy3d/components/geometry/primitives.py

```diff
@@ -5,23 +5,30 @@
 from math import isclose
 from typing import List
 
+import autograd.numpy as anp
 import numpy as np
 import pydantic.v1 as pydantic
 import shapely
 
 from ...constants import LARGE_NUMBER, MICROMETER
 from ...exceptions import SetupError, ValidationError
 from ...packaging import verify_packages_import
+from ..autograd import AutogradFieldMap, TracedSize1D
+from ..autograd.derivative_utils import DerivativeInfo
 from ..base import cached_property, skip_if_fields_missing
 from ..types import Axis, Bound, Coordinate, MatrixReal4x4, Shapely, Tuple
 from . import base
+from .polyslab import PolySlab
 
 # for sampling conical frustum in visualization
 _N_SAMPLE_CURVE_SHAPELY = 40
 
 # for shapely circular shapes discretization in visualization
 _N_SHAPELY_QUAD_SEGS = 200
 
+# number of points to discretize polyslab used to represent cylinder in `Cylinder.to_polyslab()`
+N_PTS_CYLINDER_POLYSLAB = 51
```

Comment:
Note some of the previous constants are private `_N...`, maybe this one should be too?

Reply:
sure I can do that